src/wallet/mod.rs

```diff
@@ -1158,6 +1160,45 @@ where
         builder: TxBuilder<D, Cs, Ctx>,
     ) -> Result<PSBT, Error> {
         let mut psbt = PSBT::from_unsigned_tx(tx)?;
+
+        if builder.add_global_xpubs {
+            let mut all_xpubs = self.descriptor.get_extended_keys()?;
+            if let Some(change_descriptor) = &self.change_descriptor {
+                all_xpubs.extend(change_descriptor.get_extended_keys()?);
+            }
+
+            for xpub in all_xpubs {
+                let serialized_xpub = base58::from_check(&xpub.xkey.to_string())
+                    .expect("Internal serialization error");
+                let key = PSBTKey {
+                    type_value: 0x01,
+                    key: serialized_xpub,
+                };
+
+                let (origin_fingerprint, origin_path) = match xpub.origin {
+                    Some(origin) => origin,
+                    None if xpub.xkey.depth == 0 => {
+                        (xpub.root_fingerprint(&self.secp), vec![].into())
+                    }
+                    _ => return Err(Error::MissingKeyOrigin(xpub.xkey.to_string())),
+                };
+                let mut value = origin_fingerprint.to_bytes().to_vec();
+                let path: Vec<u8> = origin_path
+                    .into_iter()
+                    .map(|child| match child {
+                        ChildNumber::Normal { index } => index.to_le_bytes(),
+                        ChildNumber::Hardened { index } => (index + 0x80000000).to_le_bytes(),
+                    })
+                    .fold(vec![], |mut concat, child| {
+                        concat.extend(&child);
+                        concat
+                    });
+                value.extend(path);
```

Comment:
I think you can use `KeySource::serialize()` instead

Reply:
That's nice, I didn't know you could serialize it directly. I can't find where they document how the child numbers are serialized,  whether they are encoded as big endian or little endian. I'll try changing the code and if the tests pass I guess it means that they use le encoding

Reply:
Looks like it's working, it looks much better this way!